kmir/src/kmir/kdist/mir-semantics/rt/data.md

```diff
@@ -433,17 +431,7 @@ The `Value` sort above operates at a higher level than the bytes representation
   rule #decodeConstant(_, _) => Any [owise]
 ```
 
-Constant operands are simply decoded according to their type.
-
-```k
-  rule <k> #readOperand(operandConstant(constOperand(_, _, mirConst(KIND, TY, _))))
-        =>
-           typedLocal(#decodeConstant(KIND, {TYPEMAP[TY]}:>RigidTy), TY, mutabilityNot)
-        ...
-      </k>
-      <basetypes> TYPEMAP </basetypes>
-    requires TY in_keys(TYPEMAP)
-```
```

Comment:
`#decodeConstant` struck me as a high-level specific function, and that `#readOperand` could be implemented in a different way in the low-level module. But this pattern looks perfectly adequate.

Reply:
Yes, we could probably refactor this to pull `#readOperand` into the interface module `RT-DATA`. But I think the current division might be better - `#readOperand` will get a lot more code to deal with  the projections.